### PR TITLE
Runner step seperation

### PR DESCRIPTION
**Pull request by lehmann-4178656ch**
* Reworked code so that formaters are happy.
* Improved `-i` behaviour
* Added example for changed behaviour

### CODE REVIEW

File: meta.go [hunk 1]
```diff
@@ -9,23 +9,34 @@ import (
...
 	KeepAliveYes ServiceKeepAlive = iota
+	// KeepAliveNo signals that the service will be killed when gantry stops.
 	KeepAliveNo
+	// KeepAliveReplace signals that the service is killed prior to replacement.
 	KeepAliveReplace
 )
 const (
+	// LogHandlerStdout signals that the standard location (stdout or stderr) is used.
 	LogHandlerStdout ServiceLogHandler = iota
+	// LogHandlerFile signlas that the log shall be stored in a file.
```
Comment (niklas88) [inline meta.go]: typo

**Comment by niklas88** [comment]
Overall I really like these changes but I think the example needs better explanations and could be in its own PR

File: examples/partial_execution/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Partial execution
+This example expands on the selective run example.
+
+The structure of the example is visualized in [pipeline.svg](./pipeline.svg).
+
+Run without arguments the example starts a `service`, waits for its
```
[the remark this answers is not in the document]
Comment (lehmann-4178656ch) [reply examples/partial_execution/README.md]: Changed.

File: examples/partial_execution/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Partial execution
+This example expands on the selective run example.
```
[the remark this answers is not in the document]
Comment (lehmann-4178656ch) [reply examples/partial_execution/README.md]: Dropped.

File: examples/partial_execution/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+The structure of the example is visualized in [pipeline.svg](./pipeline.svg).
+
+Run without arguments the example starts a `service`, waits for its
+availability and then executes a number of tests which are sequenced but
+independent.
+
+To skip `test_1` and `test_0` the following command should be used:
+```
+gantry -i test_0 -i test_1
```
[the remark this answers is not in the document]
Comment (lehmann-4178656ch) [reply examples/partial_execution/README.md]: Will be moved into own PR

File: log.go [hunk 5]
```diff
@@ -136,12 +133,14 @@ func (c *ColorStore) NextColor() int {
 	return c.colors[c.index]
 }
 
+// PrefixedWriter is a writer which prefixes all lines.
```
[the remark this answers is not in the document]
Comment (lehmann-4178656ch) [reply log.go]: Changed

File: log.go [hunk 6]
```diff
@@ -150,6 +149,7 @@ func NewPrefixedWriter(prefix string, target io.Writer) *PrefixedWriter {
 	}
 }
 
+// Write writes bytes to a internal buffer and outputs the data to the internal target.
```
[the remark this answers is not in the document]
Comment (lehmann-4178656ch) [reply log.go]: Thanks, I always make these dumb mistakes.

File: meta.go (hunk 1 above)
Comment (lehmann-4178656ch) [reply meta.go]: fixed

File: log.go [hunk 7]
```diff
@@ -174,33 +175,44 @@ func (p *PrefixedWriter) Output() error {
...
 	return &PrefixedLogger{
 		prefix: prefix,
 		logger: logger,
 	}
 }
 
+// Printf format prints to the logger.
 func (p *PrefixedLogger) Printf(format string, v ...interface{}) {
-	p.logger.Output(2, fmt.Sprintf(PrefixedWriterFormat, p.prefix, fmt.Sprintf(format, v...)))
+	if err := p.logger.Output(2, fmt.Sprintf(PrefixedWriterFormat, p.prefix, fmt.Sprintf(format, v...))); err != nil {
```
[the remark this answers is not in the document]
Comment (lehmann-4178656ch) [reply log.go]: I use the same code as the golang build-in log package:
https://github.com/golang/go/blob/d9b13233378668a4fd24ac5d044e9d550cd2a8be/src/log/log.go#L178-L180

**Comment by lehmann-4178656ch** [comment]
Thank you for your comments, the log topic will be reworked soon with a different PR

**Comment by lehmann-4178656ch** [comment]
Thanks again for your time!

**Pull request merged**